addons/netfox/rollback/rollback-synchronizer.gd

```diff
@@ -222,8 +236,16 @@ func history_cleanup() -> void:
 		_inputs.erase(_inputs.keys().min())
 		
 	if (NetworkRollback.enable_input_serialization):
-		while _serialized_inputs.size() > NetworkRollback.history_limit:
-			_serialized_inputs.erase(_serialized_inputs.keys().min())
+		if (NetworkRollback.serialized_inputs_history_limit > 0):
```

Comment:
0 or -1 value is for saving all states/inputs, which is required for saving replays (well, the inputs at least, but now states are required instead until that `.fire()` issue is solved)